### ISSUE TITLE
Error: $ operator not defined for this S4 class in reduceDimension

### ISSUE THREAD

**Issue by Kekananen** [cole-trapnell-lab/monocle-release#425]
Hi, I'm following Tutorial II and everything appears to run without error up until running the command:
`MCA_cds <- reduceDimension(MCA_cds, max_components = 2,
                       reduction_method = 'UMAP',
                       metric="correlation",
                       min_dist = 0.75,
                       n_neighbors = 50,
                       verbose = T)`
in which the error is thrown as follows:
`Error: $ operator not defined for this S4 class in reduceDimension`

If anyone has any advice on how to move past this that would be good before I start molding my own data.  It's the same for Tutorial I (same error in the reduceDimension function), but I what I'm really after is a graph much like the one in the second one.  I'm using R 4.0.0 is that's of any concern in this problem.  Thanks in advance.

**Comment by July233** [comment]
me too...you solved it?

**Comment by Kekananen** [comment]
No, I unfortunately gave up on this since I couldn’t move past it. I do hope in the future this gets fixed though since it’s an awesome tool. My only guess is I had something incompatible in my version when I installed something.

**Comment by MackZhang** [comment]
I have same problem

**Comment by razorofockham** [comment]
Are you guys by any chance running `monocle3_alpha`? I am asking because I ran into the same error upon switching from R 3.6.3 to R 4.1.2, and after some debugging I managed to get rid of it by replacing line 642 in `order_cells.R` (see [here](https://github.com/cole-trapnell-lab/monocle-release/blob/dd61e47bf1848a7334a3e9806b0eae25d25a4800/R/order_cells.R#L642C15-L642C158)) with the following:

`adj_mat <- Matrix::sparseMatrix(i = tmp$graph_@j, p = tmp$graph_@p, x = -as.numeric(tmp$graph_@x), dims = c(ncol(cds), ncol(cds)), index1 = F, dimnames = list(colnames(cds), colnames(cds)))`

I believe this may have had something to do with `reticulate` being updated upon upgrading R, which in turn would lead to the variable `tmp` (the output of calling UMAP within `reticulate`) not having the same structure as before the update.
